src/chips/sam4l/dma.rs

```diff
@@ -134,7 +134,7 @@ pub struct DMAChannel {
 }
 
 pub trait DMAClient {
-    fn xfer_done(&mut self);
+    fn xfer_done(&mut self, pid: usize);
```

Comment:
The `pid` is 8-bits in the hardware, so probably should be a `u8` here (should also, then, change `do_xfer`).